src/binder.ts

```diff
@@ -410,6 +410,19 @@ export class Binder implements IDisposable {
     const debugAdapter = new DebugAdapter(dap, this._asyncStackPolicy, launchParams, container);
     const thread = debugAdapter.createThread(cdp, target);
 
+    const isBlazor = 'inspectUri' in launchParams && !!launchParams.inspectUri;
+    if (isBlazor) {
+      const telemetryReporter = this._rootServices.get(ITelemetryReporter) as ITelemetryReporter;
+      const isVsCode = this._rootServices.get(IsVSCode);
+      cdp.DotnetDebugger.on('reportBlazorDebugError', event => {
+        telemetryReporter.report('blazorDebugError', {
+          exceptionType: event.exceptionType,
+          '!error': event.error,
+          error: isVsCode ? undefined : event.error,
+        });
+      });
+    }
```

Comment:
Eventually I was to have some kind of `IDebuggerPlugin { register(thread, cdp, dap): void }` interface, but for now can we toss this in its own small method or class?

Reply:
Moved to event listener addition to a helper method. Is binder the right place?

I also added the condition to check if it was blazor debugging based on https://github.com/microsoft/vscode-js-debug/blob/31cdfeb4da0b8c221790bc36df54087614c8b115/src/targets/sourcePathResolverFactory.ts#L81, since otherwise it seemed like any other CDP connection would be able to emit the event and send a telemetry report.